app/src/main/java/com/example/moneymanager/MainActivity.kt

```diff
@@ -34,30 +34,25 @@ class MainActivity : AppCompatActivity() {
         }
 
         // set the id for the progressbar and progress text
-        createTransactionList()
         val transactionRecycler = findViewById<RecyclerView>(R.id.rvTransaction)
-        transactionRecycler.layoutManager = LinearLayoutManager(this, RecyclerView.VERTICAL,
-                false)
-        val firstSection = TransactionAdapter(transactionList)
+        transactionRecycler.layoutManager = LinearLayoutManager(
+            this, RecyclerView.VERTICAL,
+            false
+        )
+        val firstSection = TransactionAdapter(transactions)
         transactionRecycler.adapter = firstSection
     }
 
-    private fun createTransactionList() {
-        for ((index, names) in transactionNames.withIndex()) {
-            val transactionList = TransactionItem(transactionLogos[index], names,
-                    transactionDates[index], budget[index])
-            this.transactionList.add(transactionList)
-        }
-    }
-
     private fun showBottomSheetDialog() {
         val bottomSheetDialog = BottomSheetDialog(this)
         bottomSheetDialog.setContentView(R.layout.bottom_sheet_dialog)
         bottomSheetDialog.setCanceledOnTouchOutside(false)
         val btnCreate = bottomSheetDialog.findViewById<Button>(R.id.btnCreate)
         btnCreate!!.setOnClickListener {
```

Comment:
It is safer (and advisable, I suppose) to use the null check operator `?` than the double bang `!!`, unless you're 100% sure that the receiver is not going to be null. Even then, you probably should play it safe and use the null check operator instead: Consider changing to the below, which uses the null check operator:

`btnCreate?.setOnClickListener {
    ...
}`